### ISSUE TITLE
Fetch old InventoryEntries without reference expansion

### ISSUE THREAD

**Issue by jakubchrobak** [commercetools/commercetools-sync-java#55]
### Description
As @heshamMassoud mentioned, reference expansion is quite inefficient. Currently query of old inventories requires reference expansion so that supply channel `key` could be extracted. It could be easily replaced by id-key mapping, so that we could obtain `key` having only `id` in not expanded reference. At the beginning of inventory sync we cache opposite mapping of key-to-id so we could also store id-to-key.
Efficiency of the syncer should increase then.

### Expected Behaviour
Fetching old inventory entries from CTP requires supply channels reference expansion.

### Current Behaviour
Fetching old inventory entries from CTP should not requiressupply channels reference expansion.

### Possible Solution
I would create a wrapper that would represents existing channels.
It would be backed on two HashMaps. 
It would allow to get id for key and key for id of supply channels. 
The wrapper would be more convenient than separate HashMaps, because as @piobra suggested we should return actual results from sync methods, instead of modify global sync variables.

**Comment by heshamMassoud** [comment]
@jakubchrobak 
> Currently query of old inventories requires reference expansion so that supply channel key could be extracted. 

but why would you need the key of the channel referenced on the old inventory? 

The sync already does reference resolution on the new inventoryEntries, so that they have correct ids on their references. So now at this point you have both the old and new InventoryEntry with the id on the reference ready for comparison.

**Comment by jakubchrobak** [comment]
@heshamMassoud Yes, you're right! I was so focused on sku and **key** comparision that I didn't noticed that.

**Comment by heshamMassoud** [comment]
@jakubchrobak I opened the issue again, since issues are only closed when they are in the release :)

**Comment by jakubchrobak** [comment]
@heshamMassoud 
 Ah, sorry. In fact the issue closed itself automatically after I had merged the relevant PR. Maybe it can be disbled somewhere in the repo settings?
